### DISCUSSION TITLE
Customize IDH

### DISCUSSION THREAD

**Discussion by izidood** [Security-Onion-Solutions/securityonion#12484] [2.4] [answered]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

32

### Storage for /

100GB

### Storage for /nsm

500GB

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi.
So i want to make Teams notifications for the IDH's.
i know what to do, and ive gotten it to work, but made it in the default.yaml which will make it dissappear whenever the node is restarted which makes sense. So is there a path where i can make a copy of the IDH default.yaml to, or how should i approach this?. I dont see anywhere that i can add a Teams notification setup in the UI sadly, so i assume i have to put it directly in the file.

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by m0duspwnens** [comment, marked as answer]
If you made changes to the defaults.yaml under /opt/so/saltstack/default/salt/idh, then it will survive reboots, but your changes will be lost when Security Onion is upgraded. In order for the changes to persist through upgrades, then we need to add it to SOC. Since this option isn't exposed in SOC by default, then we have to add it via the advanced options:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/17089008/20f3dbb9-6ce2-4203-9029-1ff47f83f9b8)

**Reply by jaypfe** [reply]
A potential other solution to use here that I have implemented is to build an elastalert rule that looks in ".ds-logs-playbook.alerts-so*"
with a query like "event_data.data_stream.dataset:idh" 

Then use the Teams alerter that is native to elastalert to fire your alerts. So whenever an alert fires it will drop into that index and your rule will see it and alert you. This is how I alert on all of types of rules in SOC and Kibana. 

You can include the URL for the alert as well if you have one generate in the logs, or include a custom generated Kibana dashboard for it

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/107776997/c23fe2c4-ba54-4569-94a5-e4efdc7b7318)


https://elastalert.readthedocs.io/en/latest/ruletypes.html#common-configuration-options

**Comment by izidood** [comment]
Fantastic. Thank you guys!
